pkg/deviceshifu/deviceshifubase/pushtelemetry.go

```diff
@@ -110,6 +118,49 @@ func pushToShifuTelemetryCollectionService(message *http.Response, request *v1al
 	return nil
 }
 
+func getSecret(c *rest.RESTClient, name, ns string) (map[string]string, error) {
+	secret := v1.Secret{}
+	secretPath := fmt.Sprintf("/api/v1/namespaces/%s/%s/%s", ns, SecretResource, name)
+	if err := c.Get().AbsPath(secretPath).Do(context.TODO()).Into(&secret); err != nil {
+		return nil, err
+	}
+	res := make(map[string]string)
+	for k, v := range secret.Data {
+		res[k] = string(v)
+	}
+	return res, nil
+}
+
+func injectSecret(c *rest.RESTClient, ts *v1alpha1.TelemetryService, ns string) {
+	if ts.Spec.ServiceSettings == nil {
+		logger.Warn("empty telemetry service setting.")
+		return
+	}
+	if ts.Spec.ServiceSettings.HTTPSetting == nil {
+		logger.Info("service setting is not HTTP, skip secret injection")
```

Comment:
logger.Warn

Reply:
All telemetry will be scanned. Deviceshifubase only inject secret into HTTP setting and all others must be warned?